src/commands/targets.rs

```diff
@@ -0,0 +1,197 @@
+use anyhow::{bail, Context, Result};
+use clap::Args;
+use std::{
+    fs,
+    path::{Path, PathBuf},
+};
+use wac_types::{ExternKind, ItemKind, Package, SubtypeChecker, Types, WorldId};
+
+/// Parses a WAC source file into a JSON AST representation.
+#[derive(Args)]
+#[clap(disable_version_flag = true)]
+pub struct TargetsCommand {
+    /// The path to the component
+    #[clap(value_name = "COMPONENT_PATH")]
+    pub component: PathBuf,
+    /// The path to the WIT definition
+    #[clap(value_name = "WIT_PATH")]
+    pub wit: PathBuf,
+    /// The name of the world to target
+    ///
+    /// If the wit package only has one world definition, this does not need to be specified.
+    #[clap(long)]
+    pub world: Option<String>,
+}
+
+impl TargetsCommand {
+    /// Executes the command.
+    pub async fn exec(self) -> Result<()> {
+        log::debug!("executing targets command");
+        let mut types = Types::default();
+
+        let wit_bytes = encode_wit_as_component(&self.wit)?;
+        let wit = Package::from_bytes("wit", None, wit_bytes, &mut types)?;
+
+        let component_bytes = fs::read(&self.component).with_context(|| {
+            format!(
+                "failed to read file `{path}`",
+                path = self.component.display()
+            )
+        })?;
+        let component = Package::from_bytes("component", None, component_bytes, &mut types)?;
+
+        let wit = get_wit_world(&types, wit.ty(), self.world.as_deref())?;
+
+        validate_target(&types, wit, component.ty())?;
+
+        Ok(())
+    }
+}
+
+/// Gets the selected world from the component encoded WIT package
+fn get_wit_world(
+    types: &Types,
+    top_level_world: WorldId,
+    world_name: Option<&str>,
+) -> anyhow::Result<WorldId> {
+    let top_level_world = &types[top_level_world];
+    let world = match world_name {
+        Some(world_name) => top_level_world
+            .exports
+            .get(world_name)
+            .with_context(|| format!("wit package did not contain a world name '{world_name}'"))?,
```

Comment:
I think we use backticks in most of our error messages to delimit names.

```suggestion
            .with_context(|| format!("wit package did not contain a world named `{world_name}`"))?,
```